diff --git a/dist/module/GoTrueClient.js b/dist/module/GoTrueClient.js
index 91c28b83361c4cc91c1d6d573091616055b220cd..1b2ee8a20e431b56b0ca15f39cf1d5bac2838143 100644
--- a/dist/module/GoTrueClient.js
+++ b/dist/module/GoTrueClient.js
@@ -12,6 +12,7 @@ const DEFAULT_OPTIONS = {
     url: GOTRUE_URL,
     storageKey: STORAGE_KEY,
     autoRefreshToken: true,
+    broadcast: true,
     persistSession: true,
     detectSessionInUrl: true,
     headers: DEFAULT_HEADERS,
@@ -63,6 +64,7 @@ export default class GoTrueClient {
             this.logger = settings.debug;
         }
         this.persistSession = settings.persistSession;
+        this.broadcast = settings.broadcast;
         this.storageKey = settings.storageKey;
         this.autoRefreshToken = settings.autoRefreshToken;
         this.admin = new GoTrueAdminApi({
@@ -103,7 +105,7 @@ export default class GoTrueClient {
             this.memoryStorage = {};
             this.storage = memoryLocalStorageAdapter(this.memoryStorage);
         }
-        if (isBrowser() && globalThis.BroadcastChannel && this.persistSession && this.storageKey) {
+        if (isBrowser() && globalThis.BroadcastChannel && this.persistSession && this.storageKey && this.broadcast) {
             try {
                 this.broadcastChannel = new globalThis.BroadcastChannel(this.storageKey);
             }
